### ISSUE TITLE
store.when("product").updated(function (p) {}); not firing Android 4.2.2

### ISSUE THREAD

**Issue by fer8a** [j3k0/cordova-plugin-purchase#142]
Hello @j3k0,

I'm using this plugin within telerik mobile platform (http://plugins.telerik.com/plugin/inapppurchase) and I'm having an issue to make the above named function to work.

I'm testing with a Samsung Galaxy s4 mini Android v4.4.2 and everything works fine (attached screenshots for logs)
![samsung galaxy s4 mini 1](https://cloud.githubusercontent.com/assets/3921419/6464542/1e87d7ac-c18d-11e4-8d89-d44e42ec2a2d.png)
![samsung galaxy s4 mini 2](https://cloud.githubusercontent.com/assets/3921419/6464543/1e89e452-c18d-11e4-994d-398c2c3e7db8.png)

first screenshot shows my calls to store.register({ }); and the second one shows the call to store.when("product").updated(function (p) {});

All good so far, but when I test my app in a Blu Dash 4.0 Music Android v4.2.2 the update function doesn't fire (attached screenshot with all the logs ouputed for this device)
![blu dash 4 music](https://cloud.githubusercontent.com/assets/3921419/6464634/c0ba999c-c18d-11e4-9b60-0398b4d44e43.png)

My code goes something like this:

```javascript
store.register({
   id: 'dreams.crow.1',
   alias: 'crow',
   type: store.NON_CONSUMABLE
});
...
store.when("product").updated(function (p) {
  if (p.valid) {
    //logic
    if (p.canPurchase) {
      //logic
    }
    if (p.owned) {
      //logic
    }
    console.log('modified');
  }
});
```
Any hints about why this might be happening would be very welcome.

**Comment by Heidschnucke** [comment]
Are you sure, that "updated" is not called? Because you only do the console.log if the product is valid.
So I think, your products are not valid.

A thing that the guide here is not mentioning is, that the google play store changed the behaviour for test apps. You have to upload the (signed) apk to the play store for alpha or beta state, add the inapp purchases and then publish it. 

The publishing wasn't necessary yet. It is now. Without that, that purchases are never valid.

**Comment by fer8a** [comment]
Hello @Heidschnucke,

Yes, they are indeed valid as you can see in my second attached screenshot. Also the apk is already in beta release on Google Play and using my Samsung Galaxy s4 mini I'm able to go through all the purchase proccess with no problems.

But using the other device (Blu Dash) this is not happening. Just in case I console.log another message outside the p.valid conditional like you suggested but still nothing happens.

**Comment by Heidschnucke** [comment]
Hello @fer8a,

ok. I had a similiar problem, so I thought it could fit for you too.

Cordovas "device ready" isn't fired. Just another guess, but maybe this is the crux of the matter.
Docs say: "As with any other plugin, this object shouldn't be used before the "deviceready" event is fired."

**Comment by fer8a** [comment]
It's ok, thanks for the input though any help is appreciated.

The "deviceready" event is also fired normally. I just happen to wrap my app within a requirejs module and the log you see there is just telling me that is taking longer than 5 seconds to initialize (something I nee to work on :p) but is properly working.

**Comment by fer8a** [comment]
Turns out the problem was due to the debugger plugin I'm using (jshybugger-plugin-4.5.7). Once I removed it everything started to work as it should.

So for further references if anyone is trying to remote debug with this plugin for Android devices with version prior to 4.4 use the version 3.4.1 (jshybugger-plugin-3.4.1) further releases will just break your application.

Closing this issue, thanks for the awesome plugin @j3k0
